### DISCUSSION TITLE
Metabase Integration?

### DISCUSSION THREAD

**Discussion by GabrielCeffas** [outline/outline#2437] [Ideas]
Hi, my company uses Metabase a lot for our Analytics. We would like to be able to embed some reports or dashboards inside some wiki pages. Is this possible or something that could be implemented? [Here](https://www.metabase.com/docs/latest/administration-guide/13-embedding.html) is a link to Metabase's embed documentation.

**Comment by tommoor** [comment]
We also use Metabase here at Outline, agreed it would be really nice to have :) 

The unfortunate thing is our embeds work by initially matching against a domain currently – this isn't a great strategy for self-hosted integrations as obviously the domain is different for each install/team. I think it could be achieved by having somewhere in the admin that you can add your Metabase domain though or through generic iframe functionality which has also been suggested

**Reply by GabrielCeffas** [reply]
Either way would be great honestly! Of course generic iframes would be great for other applications as well. Looking forward to seeing future progress on this! :)

**Reply by tmvy** [reply]
> The unfortunate thing is our embeds work by initially matching against a domain currently – this isn't a great strategy for self-hosted integrations as obviously the domain is different for each install/team.

Would it be possible to match by an additional unique query string parameter, rather than domain, for sites such as Metabase?

I don't know enough front-end to understand whether this would be suitable approach - but if it is, I could imagine it being a good solution for those comfortable with modifying URLs, while we await a more friendly UI-based solution.

For example `https://anywebsite.com/page?outline_embed=metabase`, where `outline_embed=metabase` is the unique parameter for Outline to match, assuming that most sites ignore such additional parameters once the link is followed. The author of the document containing the embed would append this to their link before adding it to the document.

**Reply by tommoor** [reply]
Looking back on this I believe it could be configurable without any extra steps, just gotta find the time to build it out 😅
